hack/verify-golangci-lint.sh

```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.44.0
+VERSION=v1.46.2
```

Comment:
Had to bump this up due to https://github.com/golangci/golangci-lint/issues/2374